src/rpc/rawtransaction.cpp

```diff
@@ -1470,6 +1470,139 @@ UniValue sendrawtransaction(const UniValue &params, bool fHelp)
+{
+    std::map<std::string, std::string>::iterator it;
+
+    result.pushKV("isValid", input.isValid);
+    UniValue uv_vdata(UniValue::VARR);
+    for (auto data : input.vData)
+    {
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("isValid", data.isValid);
+        UniValue entry_metadata(UniValue::VOBJ);
+        for (it = data.metadata.begin(); it != data.metadata.end(); ++it)
+        {
+            entry_metadata.pushKV(it->first, it->second);
+        }
+        entry.pushKV("metadata", entry_metadata);
+        UniValue entry_errors(UniValue::VARR);
+        for (auto error : data.errors)
+        {
+            entry_errors.push_back(error);
+        }
+        entry.pushKV("errors", entry_errors);
+        uv_vdata.push_back(entry);
+    }
+    result.pushKV("inputs", uv_vdata);
+}
+
+UniValue validaterawtransaction(const UniValue &params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 3)
+    {
+        throw std::runtime_error(
+            "validaterawtransaction \"hexstring\" ( allowhighfees, allownonstandard )\n"
+            "\nValidates raw transaction (serialized, hex-encoded) to local node without broadcasting it.\n"
+            "\nAlso see createrawtransaction and signrawtransaction calls.\n"
+            "\nArguments:\n"
+            "1. \"hexstring\"    (string, required) The hex string of the raw transaction)\n"
+            "2. allowhighfees    (boolean, optional, default=false) Allow high fees\n"
+            "3. allownonstandard (string 'standard', 'nonstandard', 'default', optional, default='default')\n"
+            "                    Force standard or nonstandard transaction check\n"
+            "\nResult:\n"
+            "\"hex\"             (string) The transaction hash in hex\n"
+            "\nExamples:\n"
+            "\nCreate a transaction\n" +
+            HelpExampleCli("createrawtransaction",
+                "\"[{\\\"txid\\\" : \\\"mytxid\\\",\\\"vout\\\":0}]\" \"{\\\"myaddress\\\":0.01}\"") +
+            "Sign the transaction, and get back the hex\n" + HelpExampleCli("signrawtransaction", "\"myhex\"") +
+            "\nSend the transaction (signed hex)\n" + HelpExampleCli("sendrawtransaction", "\"signedhex\"") +
+            "\nAs a json rpc call\n" + HelpExampleRpc("validaterawtransaction", "\"signedhex\""));
+    }
+
+    RPCTypeCheck(params, boost::assign::list_of(UniValue::VSTR)(UniValue::VBOOL)(UniValue::VSTR));
+
+    // parse hex string from parameter
+    CTransaction tx;
+    if (!DecodeHexTx(tx, params[0].get_str()))
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+    CTransactionRef ptx(MakeTransactionRef(std::move(tx)));
+    const uint256 &hashTx = ptx->GetHash();
+
+    bool fOverrideFees = false;
+    TransactionClass txClass = TransactionClass::DEFAULT;
+
+    // 2nd parameter allows high fees
+    if (params.size() > 1)
+    {
+        fOverrideFees = params[1].get_bool();
```

Comment:
also check for "allowhighfees" and "disallowhighfees".  We need provide an alternative to passing true/false in placement args since this doesn't work well on the command line.  Yes I know that the original RPCs were just true/false, but we can do better.  If we ever merge in the key=value form that core adopted, it'll be still OK to also allow these 2 strings.

Reply:
added